### PR TITLE
Bug offers messaging

### PR DESCRIPTION
**Pull request by jysmys**
### BUG-FIXES
[PT link](https://www.pivotaltracker.com/story/show/173496871)
[PT link](https://www.pivotaltracker.com/story/show/173501045)
[PT link](https://www.pivotaltracker.com/story/show/173501806)

BUG- When you offer help you cannot see you offer as pending
BUG- Quest not saved for the helper
BUG-Helper can see active requests
BUG -When request card are still open --> Crash

### CODE REVIEW

**Review by ViaMarcus — APPROVED** [review]
Lots of logical changes, bit hard to follow, but I didn't spot anything blatant

**Review by kermit-klein — APPROVED** [review]

**Comment by kermit-klein** [comment]
Amazing refactoring, great work Jenny! I applaud

**Review by designerofthing — APPROVED** [review]

File: src/components/MyRequestCard.jsx [hunk 1]
```diff
@@ -15,13 +15,12 @@ const MyRequestCard = ({ request, page }) => {
     if (myActiveRequest) {
```
Comment (erikbjoern) [inline src/components/MyRequestCard.jsx]: Heads up: you haven't pushed your latest changes I see

**Review by erikbjoern — CHANGES_REQUESTED** [review]

File: src/components/MyRequestCard.jsx (hunk 1 above)
Comment (jysmys) [reply src/components/MyRequestCard.jsx]: aaa, thx!

**Review by erikbjoern — APPROVED** [review]
